server/src/main/java/org/elasticsearch/index/mapper/JsonFieldMapper.java

```diff
@@ -315,23 +315,23 @@ public Query rangeQuery(Object lowerTerm,
         @Override
         public Query fuzzyQuery(Object value, Fuzziness fuzziness, int prefixLength, int maxExpansions,
                                 boolean transpositions) {
-            throw new UnsupportedOperationException("[fuzzy] queries are not currently supported on [" +
-                CONTENT_TYPE + "] fields.");
+            throw new UnsupportedOperationException("[fuzzy] queries are not currently supported on keyed " +
+                "[" + CONTENT_TYPE + "] fields.");
         }
 
         @Override
         public Query regexpQuery(String value, int flags, int maxDeterminizedStates,
                                  MultiTermQuery.RewriteMethod method, QueryShardContext context) {
-            throw new UnsupportedOperationException("[regexp] queries are not currently supported on [" +
-                CONTENT_TYPE + "] fields.");
+            throw new UnsupportedOperationException("[regexp] queries are not currently supported on keyed " +
+                "[" + CONTENT_TYPE + "] fields.");
```

Comment:
Same here, just adding the field prefix should work ?

Reply:
Do you think it'd make sense to automatically adjust `maxDeterminizedStates` as well?

Reply:
let's keep the value as is, the field name shouldn't consume a lot of states and the current limit is already high (`10,000`).